lib/iris/cube.py

```diff
@@ -4103,17 +4113,27 @@ def aggregated_by(self, coords, aggregator, **kwargs):
             dimensions=dimension_to_groupby, dim_coords=True
         ) or [None]
         for coord in groupby.coords:
+            new_coord = coord.copy()
+
             if (
                 dim_coord is not None
                 and dim_coord.metadata == coord.metadata
                 and isinstance(coord, iris.coords.DimCoord)
             ):
-                aggregateby_cube.add_dim_coord(
-                    coord.copy(), dimension_to_groupby
-                )
+                aggregateby_cube.add_dim_coord(new_coord, dimension_to_groupby)
             else:
+                # The metadata may have changed (e.g. climatology), so check if
+                # there's a better coord to pass to self.coord_dims
+                lookup_coord = coord
+                for (
+                    cube_coord,
+                    groupby_coord,
+                ) in groupby.coord_replacement_mapping:
+                    if coord == groupby_coord:
+                        lookup_coord = cube_coord
```

Comment:
If `coord_replacement_mapping` was a dictionary, could just do

```python
lookup_coord = groupby.coord_replacement_mapping.get(coord, coord)
```

Reply:
I don't think I can do that because a coord isn't hashable so can't be used as a dictionary key?

Reply:
Huh.  Right you are.